tests-ui/tests/store/modelToNodeStore.test.ts

```diff
@@ -0,0 +1,245 @@
+import { createPinia, setActivePinia } from 'pinia'
+import { beforeEach, describe, expect, it } from 'vitest'
+
+import {
+  ModelNodeProvider,
+  useModelToNodeStore
+} from '@/stores/modelToNodeStore'
+import { type ComfyNodeDefImpl, useNodeDefStore } from '@/stores/nodeDefStore'
+
+describe('useModelToNodeStore', () => {
+  let store: ReturnType<typeof useModelToNodeStore>
+  let nodeDefStore: ReturnType<typeof useNodeDefStore>
+
+  // Create minimal mock for testing - only includes 'name' field since that's
+  // the only property ModelNodeProvider constructor uses and tests verify
+  const createMockNodeDef = (name: string): ComfyNodeDefImpl => {
+    return { name } as ComfyNodeDefImpl
+  }
+
+  beforeEach(() => {
+    setActivePinia(createPinia())
+    store = useModelToNodeStore()
+    nodeDefStore = useNodeDefStore()
+
+    const mockNodeNames = [
+      'CheckpointLoaderSimple',
+      'ImageOnlyCheckpointLoader',
+      'LoraLoader',
+      'LoraLoaderModelOnly',
+      'VAELoader',
+      'ControlNetLoader',
+      'UNETLoader',
+      'UpscaleModelLoader',
+      'StyleModelLoader',
+      'GLIGENLoader'
+    ]
+
+    const mockNodeDefs: Record<string, ComfyNodeDefImpl> = Object.fromEntries(
+      mockNodeNames.map((name) => [name, createMockNodeDef(name)])
+    )
```

Comment:
If this is going to be the same every time, it could be worth extracting as a CONSTANT.
Or if it needs to be regenerated because it is mutable/mutated, an extracted method.